openedx/core/djangoapps/xblock/utils.py

```diff
@@ -67,3 +69,32 @@ def validate_secure_token_for_xblock_handler(user_id, block_key_str, token):
     # All computations happen above this line so this function always takes a
     # constant time to produce its answer (security best practice).
     return bool(result1 or result2)
+
+
+def get_key_for_anonymous_user(user):
+    """
+    Get a unique string that uniquely identifies the current anonymous (non
+    registered) user. (This is different than the "anonymous user ID", which
```

Comment:
I think all this will become a lot clearer if we renamed all the relevant names from `anonymous_user_id` to `id_for_anonymous_user` (and then this function can be `get_id_for_anonymous_user` as well).

Reply:
I think "non-loggedin" is clearer than "non-registered".

Reply:
I've updated the wording per your suggestions - thanks.